types/react/index.d.ts

```diff
@@ -2485,7 +2489,7 @@ declare namespace React {
         noModule?: boolean | undefined;
         referrerPolicy?: HTMLAttributeReferrerPolicy | undefined;
         src?: string | undefined;
-        type?: string | undefined;
+        type?: LiteralUnion<'module' | 'importmap'>;
```

Comment:
If this is really just about https://github.com/microsoft/TypeScript/issues/29729#issuecomment-567871939, let's use the pattern described there instead of adding these complicated types.